test/parse.js

```diff
@@ -215,6 +215,16 @@ test('query strings having a brackets+separator array and format option as `brac
 	}), {foo: ['']});
 });
 
+test('query strings having a brackets+separator array and format option as `bracket-separator` with a URL encoded value', t => {
+	const key = 'foo[]';
+	const value = 'a,b,c,d,e,f'
+	t.deepEqual(queryString.parse(`?${encodeURIComponent(key)}=${encodeURIComponent(value)}`, {
```

Comment:
NOTE: I have had to encode the key and value separately as the `encodeURIComponent` function will encode the `?` and `=` characters which breaks the test. 

I attempted `encodeURI` function but this doesn't encode the `,` character which we need encoded for this test.